contracts/lib/DepositContract.sol

```diff
@@ -61,10 +66,9 @@ contract DepositContract is Initializable {
         bytes32 node = leafHash;
 
         // Avoid overflowing the Merkle tree (and prevent edge case in computing `_branch`)
-        require(
-            depositCount < _MAX_DEPOSIT_COUNT,
-            "DepositContract:_deposit: Merkle tree full"
-        );
+        if (depositCount > _MAX_DEPOSIT_COUNT) {
```

Comment:
Regarding this as mentioned in https://github.com/spearbit-audits/review-polygon-zkevm-contracts/issues/22 , this would allow for `_MAX_DEPOSIT_COUNT + 1` deposits which I think would overflow the Merkle tree?

Reply:
The bit representation for `_MAX_DEPOSIT_COUNT + 1` will be identical to the zero deposit when we look only at the right most 32 bits. But fortunately, we increment `depositCount` the first time this function is called, so we don't already use this leaf and hence, nothing should be overwritten as far as I can see.

Reply:
Yes, this was a little bit gas saving, the last deposit if i'm not mistaken will happen something like this:
depositCount = 2**32 --> the first 32 bits will be 0
```solidity
  uint256 size = ++depositCount;
        for (
            uint256 height = 0;
            height < _DEPOSIT_CONTRACT_TREE_DEPTH;
            height++
        ) {
            if (((size >> height) & 1) == 1) {
                _branch[height] = node;
                return;
            }
            node = keccak256(abi.encodePacked(_branch[height], node));
        }
```
Here the condition ` if (((size >> height) & 1) == 1)` will never be true, and therefore we will get into the assert(false)

Anyway this does not seem obvious so i will just put the equal in the revert condition

Reply:
Ye I agree!